File: frontend/src/types/transaction.ts
export interface TransactionBase {
  price: number
  payment_method: string
  installment_period?: number
}

export interface TransactionCreate extends TransactionBase {
  ticket_ids: number[]
}

export interface TransactionResponse extends TransactionBase {
  transaction_id: number
  user_id: number
  date: string
  receipt_id: string
}

export interface TransactionItem {
  transaction_item_id: number
  transaction_id: number
  ticket_id: number
}

export enum PaymentMethod {
  CREDIT_CARD = 'Credit Card',
  DEBIT_CARD = 'Debit Card',
  BANK_TRANSFER = 'Bank Transfer',
  CASH = 'Cash'
}

File: backend/app/schemas/transaction.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class TransactionBase(BaseModel):
    price: float
    payment_method: str
    installment_period: Optional[int] = None


class TransactionCreate(TransactionBase):
    ticket_ids: List[int]  # List of ticket IDs to purchase


class TransactionResponse(TransactionBase):
    transaction_id: int
    user_id: int
    date: datetime
    receipt_id: str

    class Config:
        from_attributes = True
